api/src/main/java/org/apache/unomi/api/Event.java

```diff
@@ -82,47 +83,83 @@ public Event() {
      * @param target    the target of the event if any
      * @param timestamp the timestamp associated with the event if provided
      */
-    public Event(String eventType, Session session, Profile profile, String scope, Item source, Item target, Date timestamp) {
-        super(UUID.randomUUID().toString());
-        this.eventType = eventType;
-        this.profile = profile;
-        this.session = session;
-        this.profileId = profile.getItemId();
-        this.scope = scope;
-        this.source = source;
-        this.target = target;
-
-        if (session != null) {
-            this.sessionId = session.getItemId();
-        }
-        this.timeStamp = timestamp;
+    public Event(String itemId, String eventType, Session session, Profile profile, String scope, Item source, Item target, Date timestamp) {
+        super(itemId);
```

Comment:
The whole idea of using server-generated IDs for events is that Unomi can process requests coming directly from the Internet and therefore they cannot be trusted. Allowed IDs to be client specified can potentially open the system to attacks.

Reply:
Ok so after reflection this change should be ok.